src/sentry/tasks/app_store_connect.py

```diff
@@ -146,18 +158,14 @@ def inner_refresh_all_builds() -> None:
         with sdk.push_scope() as scope:
             scope.set_tag("project", option.project_id)
             try:
-                # We are parsing JSON thus all types are Any, so give the type-checker some
-                # extra help.  We are maybe slightly lying about the type, but the
-                # attributes we do access are all string values.
-                all_sources: List[Mapping[str, str]] = json.loads(option.value)
-                for source in all_sources:
-                    try:
-                        source_id = source["id"]
-                        source_type = source["type"]
-                    except KeyError:
-                        logger.exception("Malformed symbol source")
-                        continue
-                    if source_type == appconnect.SYMBOL_SOURCE_TYPE_NAME:
-                        inner_dsym_download(option.project_id, source_id)
+                asc_configs = appconnect.AppStoreConnectConfig.from_option(option)
+                # Set the project option just once in case there are multiple App Store Connect
+                # sources to avoid redundant updates to the project option.
+                if len(asc_configs) > 0:
+                    project = Project.objects.get(pk=option.project_id)
+                    project.update_option("sentry:last_checked_appstore", datetime.now())
```

Comment:
I don't think we should update this here.  As I say above I think it makes more sense to update it during the task, after the builds have actually been fetched and updated.